### PR TITLE
Remove draft/metadata-notify-2

### PR DESCRIPTION
**Pull request by progval**
as an alternative to #558

### CODE REVIEW

File: extensions/metadata.md [hunk 1]
```diff
@@ -58,7 +57,7 @@ Server administrators can setup lists of allowed or blocked keys, and may also r
...
 
-When a channel's metadata is updated, all users in that channel who are subscribed to the changed key will receive a [`METADATA` message](#metadata-server-message) notifying them of the change. When a user's metadata is updated, all users sharing a channel with that user (and all users who have [`draft/metadata-notify-2` requested](#notifications) and are [Monitoring](https://ircv3.net/specs/extensions/monitor.html#monitor-command) that user) will receive a `METADATA` message notifying them of the change.
+When a channel's metadata is updated, all users in that channel who are subscribed to the changed key will receive a [`METADATA` message](#metadata-server-message) notifying them of the change. When a user's metadata is updated, all users either sharing a channel with that user or [monitoring](https://ircv3.net/specs/extensions/monitor.html#monitor-command) them will receive a `METADATA` message notifying them of the change.
```
Comment (jwheare) [inline extensions/metadata.md:60]:
```suggestion
When metadata is updated, users who have subscribed to the changed key will receive a [`METADATA` message](#metadata-server-message) notifying them of the change.

For channel metadata, this applies only to subscribers who are members of the updated channel. For user metadata, this applies only to subscribers who either share a channel with the updated user or are [monitoring](https://ircv3.net/specs/extensions/monitor.html#monitor-command) them.
```

Clarify who gets metadata notifications.

File: extensions/metadata.md [hunk 2]
```diff
@@ -722,7 +726,7 @@ As servers may rewrite values set by clients with `METADATA SET`, clients should
 
 This specification replaces the first `metadata` specification, by adding the following incompatible changes:
 
-* The `metadata-notify` key subscribed you to all keys. Since we have now added the [`SUB`](#metadata-sub) and [`UNSUB`](#metadata-unsub) subcommands, `metadata-notify-2` does not act in this way.
+* The `metadata-notify` key subscribed you to all keys. Since we have now added the [`SUB`](#metadata-sub) and [`UNSUB`](#metadata-unsub) subcommands, it is no longer deemed useful.
```
Comment (jwheare) [inline extensions/metadata.md:729]:
```suggestion
* The `metadata-notify` cap subscribed you to all keys. With the addition of explicit [`SUB`](#metadata-sub) and [`UNSUB`](#metadata-unsub) subcommands, this is no longer the case.
```

File: extensions/metadata.md [hunk 3]
```diff
@@ -722,7 +726,7 @@ As servers may rewrite values set by clients with `METADATA SET`, clients should
 
 This specification replaces the first `metadata` specification, by adding the following incompatible changes:
```
Comment (jwheare) [inline extensions/metadata.md:727]:
```suggestion
This specification replaces the `metadata-notify` cap defined in the deprecated [metadata-3.2](https://ircv3.net/specs/core/metadata-3.2) specification, by adding the following incompatible changes:
```

The heading of this section should also be updated, but it's not in this changeset so github won't let me add a suggestion.

**Review by jwheare — CHANGES_REQUESTED** [review]

File: extensions/metadata.md [hunk 4]
```diff
@@ -68,7 +67,7 @@ This specification depends on the [`batch`](../extensions/batch.html) capability
 
 This specification also uses the [standard replies](../extensions/standard-replies.html) framework.
 
-Clients MUST NOT request both (`metadata` or `metadata-notify`) and (`draft/metadata-2` or `draft/metadata-notify-2`). Servers MUST NOT accept these requests either.
+Clients MUST NOT request both (`metadata` or `metadata-notify`) and (`draft/metadata-2`). Servers MUST NOT accept these requests either.
```
[the remark this answers is not in the document]
Comment (jwheare) [reply extensions/metadata.md:70]: The previous deprecated incompatible spec was just `metadata-notify` unprefixed. This spec is `draft/metadata-2` and will ideally end up as `metadata` unprefixed.

There is no such spec as `draft/metadata` or `draft/metadata-notify`.

File: extensions/metadata.md (hunk 3 above)
Comment (progval) [reply extensions/metadata.md:727]: the link won't exist next time we update the website

Comment (jwheare) [reply extensions/metadata.md:729]:
```suggestion
This specification replaces an earlier deprecated `metadata-notify` specification, by adding the following incompatible changes:
```

Comment (jwheare) [reply extensions/metadata.md:729]: @progval happy with revised suggestion added above?

**Review by jwheare — APPROVED** [review]

**Pull request merged**